core/types/block.go

```diff
@@ -274,10 +283,27 @@ func (h *Header) EncodeRLP(w io.Writer) error {
 		}
 	}
 
+	// Write the optional fields. If any one of them is nil then don't write any subsequent ones as
+	// this will confuse the decoder. TODO: should we error out if we have non-nil values following
```

Comment:
+1 on this, it seems like their relationship is not strictly 1 dependent on another. Need to think about this more later